Add KSE wBX16 charger support

### Discussed in https://github.com/evcc-io/evcc/discussions/3820

<div type='discussions-op-text'>

<sup>Originally posted by **ChristianFleissner** July 12, 2022</sup>
Hallo zusammen,

ich habe bei mir eine KSE wBX16 Wallbox installiert. Diese ist über RS485 und einen entsprechenden USB-Adapter an einem Raspberry Pi angeschlossen, auf dem ich EVCC zum laufen bringen möchte.
Mein Alter ist leider schon etwas fortgeschritten so dass ich ein wenig vor der Wand stehe, wenn ich mit Hilfe der Dokumentation versuche, diese Wallbox anzulegen (wobei inzwischen einiges der allgemeinen Modbus-RTU-Dokumentation verschwunden ist).
Ich sollte eigentlich alle notwendigen Informationen inklusive der entsprechenden Register haben:

Allgemeine Bedienungsanleitung:
https://www.kse-wallbox.com/alle-downloads?file=files/images/Inhaltsbilder/Service/Alle%20Downloads/KSE%20Wallbox%20Bedienungsanleitung.pdf&cid=137

Registerset:
https://www.kse-wallbox.com/alle-downloads?file=files/images/Inhaltsbilder/Service/Alle%20Downloads/KSE%20Wallbox%20Registerset.pdf&cid=137

Anleitung für das KSE-eigene Lastmanagement (habe ich aus Platzgründen nicht genommen, da dieses nicht mit SMA Sunny Home Manager kommuniziert und ich nicht den Platz für zusätzliche Summenstromwandler im Verteilerkasten habe), da steht die Baudrate usw. drin:
https://www.kse-wallbox.com/alle-downloads?file=files/images/Inhaltsbilder/Service/Alle%20Downloads/Anleitung%20Lademanagement.pdf&cid=137

Was muss ich wo eintragen, damit das halbwegs funktionieren kann? Oder bin ich zu naiv und das wird nix?

Vielen Dank!</div>

Bitte sehr, war zwar nicht am Laden, sondern nur am klimatisieren, sollte für die Wallbox aber keinen Unterschied machen:

pi@raspberrypi:~ $ evcc charger --log trace
[main  ] INFO 2022/07/17 11:35:00 evcc 0.96 (7ce0297f)
[main  ] INFO 2022/07/17 11:35:00 using config file /home/pi/evcc.yaml
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 03 00 18 00 01 0d f8
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 83 02 d0 ee
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 03 00 14 00 03 4c 3a
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 83 02 d0 ee
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 04 00 10 00 01 39 fa
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 04 02 00 05 35 3b
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 03 00 03 00 01 7d ff
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 03 02 00 10 f5 80
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 03 00 17 00 01 3d fb
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 83 02 d0 ee
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 03 00 67 00 04 fc 23
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 83 02 d0 ee
Power:          modbus: exception '2' (illegal data address), function '3'
Current L1..L3: modbus: exception '2' (illegal data address), function '3'
Charge status:  C
Enabled:        true
Charged:        modbus: exception '2' (illegal data address), function '3'
Identifier:     modbus: exception '2' (illegal data address), function '3'
Diagnostic dump:
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 04 00 30 00 01 38 30
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 04 02 01 04 f5 6b
        Firmware:       1.4
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 04 00 0e 00 01 59 fc
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 84 02 d2 de
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 04 00 10 00 01 39 fa
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 04 02 00 05 35 3b
        State:  5
[kse   ] TRACE 2022/07/17 11:35:01 modbus: send 64 02 00 31 00 01 e1 f0
[kse   ] TRACE 2022/07/17 11:35:01 modbus: recv 64 02 01 00 bf 44
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
encoding/binary.bigEndian.Uint16(...)
        /opt/hostedtoolcache/go/1.18.3/x64/src/encoding/binary/binary.go:101
github.com/evcc-io/evcc/charger.(*KSE).Diagnose(0x41a2b60)
        /home/runner/work/evcc/evcc/charger/kse.go:234 +0x38c
github.com/evcc-io/evcc/cmd.(*dumper).Dump(0x4339d80, {0x41c27b8, 0x8}, {0x1340160, 0x41a2b60})
        /home/runner/work/evcc/evcc/cmd/dumper.go:199 +0x16dc
github.com/evcc-io/evcc/cmd.(*dumper).DumpWithHeader(0x4339d80, {0x41c27b8, 0x8}, {0x1340160, 0x41a2b60})
        /home/runner/work/evcc/evcc/cmd/dumper.go:32 +0x74
github.com/evcc-io/evcc/cmd.runCharger(0x248c9a0, {0x3fdbc60, 0x0, 0x2})
        /home/runner/work/evcc/evcc/cmd/charger.go:126 +0xc68
github.com/spf13/cobra.(*Command).execute(0x248c9a0, {0x3fdbc50, 0x2, 0x2})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x688
github.com/spf13/cobra.(*Command).ExecuteC(0x248c338)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x414
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/evcc-io/evcc/cmd.Execute()
        /home/runner/work/evcc/evcc/cmd/root.go:135 +0x20
main.main()
        /home/runner/work/evcc/evcc/main.go:33 +0x3c
pi@raspberrypi:~ $

--- a/charger/kse.go
+++ b/charger/kse.go
@@ -45,8 +45,9 @@ const (
 	kseRegActualPower         = 0x18 // Aktuelle Ladeleistung (W)
 	kseRegFirmwareVersion     = 0x30 // Firmware Version
 	kseRegRFIDinstalled       = 0x31 // RFID-Leser vorhanden
-	// kseRegRelayMode           = 0x35 // Umschalten 1 phasiges oder 3 phasiges Laden
 	kseRegNFCTransactionID    = 0x67 // Tag ID (8 Bytes)
+	// kseRegRelayMode           = 0x35 // Umschalten 1 phasiges oder 3 phasiges Laden
+
 )
 
 func init() {

--- a/charger/kse.go
+++ b/charger/kse.go
@@ -45,7 +45,7 @@ const (
 	kseRegActualPower         = 0x18 // Aktuelle Ladeleistung (W)
 	kseRegFirmwareVersion     = 0x30 // Firmware Version
 	kseRegRFIDinstalled       = 0x31 // RFID-Leser vorhanden
-	kseRegRelayMode           = 0x35 // Umschalten 1 phasiges oder 3 phasiges Laden
+	// kseRegRelayMode           = 0x35 // Umschalten 1 phasiges oder 3 phasiges Laden
 	kseRegNFCTransactionID    = 0x67 // Tag ID (8 Bytes)
 )
 
@@ -202,18 +202,18 @@ func (wb *KSE) Currents() (float64, float64, float64, error) {
 	return curr[0], curr[1], curr[2], nil
 }
 
-var _ api.ChargePhases = (*KSE)(nil)
+// var _ api.ChargePhases = (*KSE)(nil)
 
-// Phases1p3p implements the api.ChargePhases interface
-func (wb *KSE) Phases1p3p(phases int) error {
-	var b uint16 = 0
-	if phases == 1 {
-		b = 1 // 1p
-	}
+// // Phases1p3p implements the api.ChargePhases interface
+// func (wb *KSE) Phases1p3p(phases int) error {
+// 	var b uint16 = 0
+// 	if phases == 1 {
+// 		b = 1 // 1p
+// 	}
 
-	_, err := wb.conn.WriteSingleRegister(kseRegRelayMode, b)
-	return err
-}
+// 	_, err := wb.conn.WriteSingleRegister(kseRegRelayMode, b)
+// 	return err
+// }
 
 // Identify implements the api.Identifier interface
 func (wb *KSE) identify() (string, error) {

--- a/templates/definition/charger/kse.yaml
+++ b/templates/definition/charger/kse.yaml
@@ -3,7 +3,7 @@ products:
   - brand: KSE
     description:
       generic: wBX16
-capabilities: ["1p3p", "rfid"]
+capabilities: ["rfid"]
 requirements:
   evcc: ["sponsorship"]
 params:

--- a/templates/docs/charger/kse_0.yaml
+++ b/templates/docs/charger/kse_0.yaml
@@ -1,7 +1,7 @@
 product:
   brand: KSE
   description: wBX16
-capabilities: ["1p3p", "rfid"]
+capabilities: ["rfid"]
 requirements: ["sponsorship"]
 render:
   - default: |

